etcdserver/v3_server.go

```diff
@@ -135,16 +171,38 @@ func (s *EtcdServer) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse
 	var err error
 
 	if isTxnSerializable(r) {
-		user, err := s.usernameFromCtx(ctx)
-		if err != nil {
-			return nil, err
+		for {
+			authInfo, err := s.authInfoFromCtx(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			hdr := &pb.RequestHeader{}
+			if authInfo != nil {
+				hdr.Username = authInfo.Username
+				hdr.Revision = authInfo.Revision
+			}
+
+			result = s.applyV3.Apply(&pb.InternalRaftRequest{Header: hdr, Txn: r})
+
+			if result.err != nil {
```

Comment:
Sorry for that, I accidentally removed the check. I'll recover it in the next update.